RNTester/js/CameraRollView.js

```diff
@@ -25,105 +23,123 @@ var {
-  renderImage: PropTypes.func,
+  renderImage: GetPhotosEdge => React.Node,
 
   /**
    * imagesPerRow: Number of images to be shown in each row.
    */
-  imagesPerRow: PropTypes.number,
+
+  imagesPerRow: number,
 
   /**
    * The asset type, one of 'Photos', 'Videos' or 'All'
    */
-  assetType: PropTypes.oneOf(['Photos', 'Videos', 'All']),
-};
-
-var CameraRollView = createReactClass({
-  displayName: 'CameraRollView',
-  propTypes: propTypes,
-
-  getDefaultProps: function(): Object {
+  assetType: 'Photos' | 'Videos' | 'All',
+|}>;
+
+type State = {|
+  assets: Array<GetPhotosEdge>,
+  lastCursor: ?string,
+  noMore: boolean,
+  loadingMore: boolean,
+  dataSource: ListViewDataSource,
+|};
+
+class CameraRollView extends React.Component<Props, State> {
+  static defaultProps = {
+    groupTypes: 'SavedPhotos',
+    batchSize: 5,
+    imagesPerRow: 1,
+    assetType: 'Photos',
+    renderImage: function(asset: GetPhotosEdge) {
+      const imageSize = 150;
+      const imageStyle = [styles.image, {width: imageSize, height: imageSize}];
+      return <Image source={asset.node.image} style={imageStyle} />;
+    },
+  };
+
+  state = this.getInitialState();
+
+  getInitialState() {
     return {
-      groupTypes: 'SavedPhotos',
-      batchSize: 5,
-      imagesPerRow: 1,
-      assetType: 'Photos',
-      renderImage: function(asset) {
-        var imageSize = 150;
-        var imageStyle = [styles.image, {width: imageSize, height: imageSize}];
-        return <Image source={asset.node.image} style={imageStyle} />;
-      },
-    };
-  },
-
-  getInitialState: function() {
-    var ds = new ListView.DataSource({rowHasChanged: this._rowHasChanged});
-
-    return {
-      assets: ([]: Array<Image>),
-      groupTypes: this.props.groupTypes,
-      lastCursor: (null: ?string),
-      assetType: this.props.assetType,
+      assets: [],
+      lastCursor: null,
       noMore: false,
       loadingMore: false,
-      dataSource: ds,
+      dataSource: new ListView.DataSource({rowHasChanged: rowHasChanged}),
     };
-  },
+  }
 
   /**
    * This should be called when the image renderer is changed to tell the
    * component to re-render its assets.
    */
-  rendererChanged: function() {
-    var ds = new ListView.DataSource({rowHasChanged: this._rowHasChanged});
+  rendererChanged() {
+    const ds = new ListView.DataSource({rowHasChanged: rowHasChanged});
     this.state.dataSource = ds.cloneWithRows(
       // $FlowFixMe(>=0.41.0)
       groupByEveryN(this.state.assets, this.props.imagesPerRow),
     );
-  },
+  }
 
-  componentDidMount: function() {
+  componentDidMount() {
     this.fetch();
-  },
+  }
 
   /* $FlowFixMe(>=0.68.0 site=react_native_fb) This comment suppresses an error
    * found when Flow v0.68 was deployed. To see the error delete this comment
    * and run Flow. */
-  UNSAFE_componentWillReceiveProps: function(nextProps: {groupTypes?: string}) {
+  UNSAFE_componentWillReceiveProps(nextProps: {groupTypes?: string}) {
```

Comment:
Can we remove this `$FlowFixMe`?

Reply:
Not sure how we can remove it.

This is the Flow error I get :
`Cannot extend property Component [1] with CameraRollView because property groupTypes is read-only in object type [2] but
writable in object type [3] in the first argument of property UNSAFE_componentWillReceiveProps.`